monix-eval/shared/src/main/scala/monix/eval/Task.scala

```diff
@@ -1916,6 +1915,40 @@ sealed abstract class Task[+A] extends TaskBinCompat[A] with Serializable {
     */
   final def uncancelable: Task[A] =
     TaskCancellation.uncancelable(this)
+
+  /** Times the `Task` execution and returns its duration and the computed value.
+    *
+    * Basic usage example:
+    *
+    * {{{
+    *   Task(1 + 1)
+    *     .timed
+    *     .flatMap { case (duration, value) =
+    *       Task.eval(Logger.info("executed in " + duration.toMillis + " ms").map(_ => value)
+    *     }
+    * }}}
+    *
+    * In the previuos example, if the initial `Task` fails, the following `flatMap` will not be executed.
+    * So, you'll not be able to time the "error path".
+    *
+    * To be able to time the execution, even if the `Task` fails, you have to call `.attempt` first:
+    *
+    * {{{
+    *   Task(1 / 0)
+    *     .attempt
+    *     .timed
+    *     .flatMap {
+    *       case (duration, Right(value)) => Task.eval(Logger.info("executed in " + duration.toMillis + " ms")).map(_ => value)
+    *       case (duration, Left(e))      => Task.eval(Logger.warn("failed in " + duration.toMillis + " ms")).flatMap(_ => Task.raiseError(e))
+    *     }
+    * }}}
+    */
+  final def timed: Task[(Duration, A)] =
```

Comment:
perhaps the type should be `Task[(FiniteDuration, A)]`?

Reply:
Could question. What's the difference ?

Reply:
Ok. Thanks